### PR TITLE
fix(electron): add contextRegExp to webpack.IgnorePlugin

### PR DESCRIPTION
**Pull request by daniellacosse**
According to the [webpack v4 docs](https://v4.webpack.js.org/plugins/ignore-plugin/), `webpack.IgnorePlugin(/Regex/);` is equivalent to `new webpack.IgnorePlugin({resourceRegExp: /Regex/});`. This is, however, not the case. It seems we must now specify the context regexp as well.

Screen Shot 2021-12-02 at 11 34 39 AM

### CODE REVIEW

**Comment by bemasc** [comment]
Can you explain why this helps?  The [WebPack source code](https://github.com/webpack/webpack/blob/c181294865dca01b28e6e316636fef5f2aad4eb6/lib/IgnorePlugin.js) looks very simple, and matches the documentation.

**Comment by daniellacosse** [comment]
> Can you explain why this helps? The [WebPack source code](https://github.com/webpack/webpack/blob/c181294865dca01b28e6e316636fef5f2aad4eb6/lib/IgnorePlugin.js) looks very simple, and matches the documentation.

Ah, you're right. This is effectively the same as just removing the IgnorePlugin entirely - I removed it and electron is found as well. Was this... not working before? I'm confused.

**Comment by daniellacosse** [comment]
If I understand correctly  contextRegExp should be set to `/node_modules/` - let's see

File: src/www/electron.webpack.js [hunk 1]
```diff
@@ -54,7 +54,8 @@ module.exports = makeConfig({
     // @sentry/electron depends on electron code, even though it's never activated
     // in the browser. Webpack still tries to build it, but fails with missing APIs.
     // The IgnorePlugin prevents the compilation of the electron dependency.
-    new webpack.IgnorePlugin({resourceRegExp: /^electron$/}),
+    new webpack.IgnorePlugin(
+        {resourceRegExp: /^electron$/, contextRegExp: /@sentry\/electron/}),
```
Comment (bemasc) [inline src/www/electron.webpack.js:58]: Should this be `/^@sentry...`?

**Comment by daniellacosse** [comment]
@fortuna is there a way to test sentry locally?

File: src/www/electron.webpack.js (hunk 1 above)
Comment (jyyi1) [reply src/www/electron.webpack.js:58]: From the [sample here](https://webpack.js.org/plugins/ignore-plugin/#example-of-ignoring-moment-locales) it seems the "context" should be ended with `$` and the "resource" should be wrapped in `^...$`.

Comment (daniellacosse) [reply src/www/electron.webpack.js:58]: Yeah - from the example - `any directories ending with 'moment' will be ignored`. `@sentry/electron` has subdirectories so we want to make sure to ignore all of them

**Comment by jyyi1** [comment]
> > Can you explain why this helps? The [WebPack source code](https://github.com/webpack/webpack/blob/c181294865dca01b28e6e316636fef5f2aad4eb6/lib/IgnorePlugin.js) looks very simple, and matches the documentation.
> 
> Ah, you're right. This is effectively the same as just removing the IgnorePlugin entirely - I removed it and electron is found as well. Was this... not working before? I'm confused.

I guess the previous syntax is not correct (it is using a raw `string` instead of the object, probably webpack will ignore that, so everything are packed into the final binary.

**Review by bemasc — APPROVED** [review]

**Review by jyyi1 — APPROVED** [review]
Thanks @daniellacosse , the code LGTM, one more thing to follow up is to make sure sentry still works.

**Comment by daniellacosse** [comment]
> @fortuna is there a way to test sentry locally?

I'll merge for now to unblock us with an eye to checking Sentry ASAP.

**Comment by fortuna** [comment]
I don't know of a way to test sentry locally. I think @alalama would use the dev key and see the reports.

**Pull request merged**